### ISSUE TITLE
weapons aren’t brought to the caserna before recruiting

### ISSUE THREAD

**Issue by benhandschuh** [jsettlers/settlers-remake#639]

**Comment by andreas-eberle** [comment]
What do you mean? Please explain the problem in more detail.

**Comment by benhandschuh** [comment]
If the production of the barracks is stopped weapons will not placed at the barracks. So if weapons are produced and you have stopped the recruitment of soldiers the weapons are not placed at the barrack.

**Comment by andreas-eberle** [comment]
This is how it is expected to work. Think of a situation when you have two barracks and want to stop the production at the barrack far back in your main land and only use the barrack at the front. In this case, no weapons should be delivered to the barrack which you stopped.

**Comment by benhandschuh** [comment]
But this produces a lack of time in recruiting. When you have just one barrack an you want to stop the recruitment of soldiers for the moment but a few minutes later you need to recruit soldiers fast. Maybe we should implement a button to give the option to start deliver weapons to a barrack when recruitment is stopped.

**Comment by andreas-eberle** [comment]
I think what you actually want is a button to stop recruiting instead of stopping the work of a barrack. Because stopping the work of a barrack means that barrack shouldn't do a thing and thus also not allocate weapons that could be used in another barrack. But there is the additional need for a way to stop recruting soldiers (e.g. when you're running out of bearers).

**Comment by tom-pratt** [comment]
I think stopping production of the barracks makes sense the way it is. Storage areas can be used to stockpile weapons so you don't restrict the production of the weapons smith.

**Comment by benhandschuh** [comment]
Hmm, but it slows down the recruitment when its needed. Sometimes you stop the work of the barracks and then you got attacked. And now you want to produce soldiers but the weapons first need to get to the barracks.

**Comment by andreas-eberle** [comment]
@benhandschuh: I agree that there should be a way to stop recruting but fill up the barrack with weapons so its ready for producing soldiers. But I think this needs to be done with a separate "stop recruting" thing...

**Comment by benhandschuh** [comment]
Can't we just add a button at each barrack which says "deliver weapons" which shows up when recruitment is stopped?

**Comment by tom-pratt** [comment]
The priority button could have a fourth state for this...

Not sure what the icon should be. Maybe a hand to imply stopped but with an arrow pointing to it to imply its still accepting items.

**Comment by benhandschuh** [comment]
Yep, thats a good idea. Or the Hand with a sword nearby?

**Comment by andreas-eberle** [comment]
I don't think it makes sense to have this "stop recruting" set for individual buildings. Whenever you have one barrack open to produce soldiers, you don't want the weapons to be stacked at another barrack. And if you want to stop recruting soldiers (e.g. because you run out of bearers), you want to stop it for all of your barracks in the current partition. So if stopping the recruitment is a building-local setting, you would have to select all you barracks (in the respecting partition) and change their mode. 

Therefore I propose to add a partition-global setting to stop recruitment.

**Comment by michaelzangl** [comment]
I personally don't think we should add this feature.

If you want to store weapons and not to recruit the soldiers immediately, you have a tactical advantage. This advantage can come at a price. Currently, you need to pay space and goods and time to build a storehouse. Storehouses are meant for storing weapons until they are needed. I don't think that we should convert barracks to storehouses.

**Comment by andreas-eberle** [comment]
@codingberlin: What's your opinion with your settlers experience?

**Comment by benhandschuh** [comment]
Well, but if you got attacked you have even less time to react against an enemy. Even with weapons placed at the barracks (like it was in the original if i remember right) you still can react just very slow to an attack.

**Comment by tom-pratt** [comment]
Yeah I'm still not convinced this feature should be added. The solution is to create a storage area and to create new residences in a timely manner to ensure you have sufficient carriers. 

Settlers is all about timing and planning to keep your economy healthy. If you accidentally create a bottleneck you should have to work your way out of it

**Comment by benhandschuh** [comment]
For exmaple: I often waited until ive the next few upgrades for my sodiers before i start to recruit soldiers. But in this tactic you need the option to recruit soldiers fast in case you got attacked. If you cant store a few weapons at the barracks you loose a lot of time which could be essential. I think this would just ad a possible more to be flexible with tactic playing. And this is what we need to do, make the game more based on not just the abvantages of each civ. Need to make i also possible to win just by the movement of your units, which is at the moment nearly impossible.

**Comment by FinnPeak** [comment]
Hello, this topic caught my eye, sorry about late spam.

I think you are clearly talking about the feature that already is in the original game - manual / automatic recruiting, which is selectable in the Settlers/Recruitment menu. If you keep this as "manual", weapons are taken into all (active) barracks but soldiers are not recruited, unless spefically ordered from the selected barracks.

I would say the original feature works just well. I allows some alteration in your early game balances in weapon production / recruiting / promotions, by letting the player to control when to mobilize army - althought it may also be a great risk sometimes.

I hope this feature will be included in the game?
